src/ALCProxy.Communication/ALCClient.cs

```diff
@@ -150,13 +150,21 @@ public object SendMethod(MethodInfo method, object[] args)
         {
             if (method == null)
                 throw new ArgumentNullException();
-            if (_serverDelegate == null) //We've called the ALC unload, so the proxy has been cut off
+            if (_server == null) //We've called the ALC unload, so the proxy has been cut off
                 throw new InvalidOperationException("Error in ALCClient: Proxy has been unloaded, or communication server was never set up correctly");
             if (args == null)
                 args = new object[] { };
 
             SerializeParameters(args, out IList<object> streams, out IList<Type> argTypes);
-            object encryptedReturn = _serverDelegate( method, streams, argTypes );
+            object encryptedReturn;
+            try
+            {
+                encryptedReturn = _serverDelegate(method, streams, argTypes);
+            }
+            catch (NullReferenceException) //Some minor failures with OSX release builds using the serverDelegate, if it fails, the backup is to call the method using Invoke directly
```

Comment:
Do we generally catch exceptions like `NullReferenceException`? Doesn't that mean there is a null ref bug that needs to be fixed?

Reply:
Ok, reading https://github.com/dotnet/corefxlab/issues/2745#issuecomment-541801971, sure.

> I'm not 100% sure exactly why this is happening, but the delegate between client and server sometimes doesn't seem to exist for the OSX release. I can PR a patch that uses the old Reflection.Invoke as a backup in case the delegate fails, which doesn't exactly solve the problem but should hopefully prevent more random failures.

Reply:
Might this represent a product bug (that needs an issue open0